tests/ui/use_self.rs

```diff
@@ -454,3 +454,10 @@ mod nested_paths {
         }
     }
 }
+
+mod issue6818 {
+    #[derive(serde::Deserialize)]
```

Comment:
I was unable to generate a proper testcase for this, so I used this instead until somebody (or myself) can figure this out.

Reply:
Why is this not a proper testcase?

Reply:
I don't know if we should rely on `serde` always producing that specific issue, they might change something in their macro that doesn't trigger a false-positive anymore.

I would have preferred if we could reproduce the specific code that causes this. I gave it another stab today without success.

Reply:
I think this is fine for now.